docs/getting_started/module.md

```diff
@@ -0,0 +1,259 @@
+# Gosoline modules
+
+We saw in the previous section, [Kernels](kernel.md), that every Gosoline application involves _Kernel_ object running one or more _Modules_ which have been added to it.
+
+![kernel_lifecycle](module_1.svg)
+
+When a _Kernel_ starts running, it will launch a goroutine for each module, and each such goroutine runs the middleware configured for that _Kernel_, along with the module itself.
+
+A Gosoline _Module_ is any structure implementing the  _Module_ interface:
+
+[embedmd]:# (../../pkg/kernel/module.go /type Module interface/ /\n}/)
+```go
+type Module interface {
+	// Run the module. If the provided context is canceled you have a
+	// few seconds (configurable with kernel.killTimeout) until your module
+	// is killed (via exit(1)). If you return from Run, it is assumed that
+	// your module is done executing and (depending on the type of your module)
+	// this might trigger a kernel shutdown. If you return an error, a kernel
+	// shutdown is also triggered.
+	Run(ctx context.Context) error
+}
+```
+
+## Module lifecycle
+
+A _Module_ can have one of these four states: _Not started_, _Running_, _Finished_, or _Cancelled_.
+
+![kernel_lifecycle](module_2.svg)
+
+- **Not Started** means that the module has been added to its kernel but the kernel itself has not yet started
+- **Running** is when the Kernel has started, and thus has started all its modules, which are now running
+- **Finished** means that a module ran and, either successfully or not, completed its work
+- **Cancelled** is when the module started running, but was stopped because the kernel stopped
```

Comment:
What is the difference between finished? Why are these stages important? After all, we only run the go routines, right?

Reply:
I don't see a single mention of a module in stage "Cancelled". What do I have to search for? And when a module returns `nil` after the context was cancelled, is it now in stage "Cancelled" or "Finished"? 

In short, do we really need to document that something first is not running, then running, then no longer running again because it is finished?